CRM/Contribute/Form/Contribution/Confirm.php

```diff
@@ -2380,7 +2371,7 @@ protected function doMembershipProcessing($contactID, $membershipParams, $premiu
     }
 
     $membershipParams = $this->getMembershipParamsFromPriceSet($membershipParams);
-    if (!empty($membershipParams['selectMembership'])) {
+    if ($this->isMembershipSelected()) {
       // CRM-12233
       $membershipLineItems = [$this->getPriceSetID() => $this->getLineItems()];;
       if ($this->_separateMembershipPayment && $this->isFormSupportsNonMembershipContributions()) {
```

Comment:
Understanding if this isn't the scope of this PR:

```
Access to deprecated property $_separateMembershipPayment of class CRM_Contribute_Form_ContributionBase:                       

use $this->isSeparateMembershipPayment() function.
```

Reply:
@jofranz yeah - I could fix that one too -